libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -312,6 +315,7 @@ constructor(
         tripSession.unregisterAllBannerInstructionsObservers()
         tripSession.unregisterAllVoiceInstructionsObservers()
         tripSession.route = null
+        navigator.reset()
```

Comment:
can we make this fix be, only moving the navigator.reset()?

The `NavigationComponentProvider` looks like a change that doesn't do anything, right?

Reply:
> The `NavigationComponentProvider` looks like a change that doesn't do anything, right?

See comment above. It's needed to mock `MapboxNativeNavigator` in order to keep `onDestroyCallsNativeNavigatorReset` test.